pkg/api/image/formatter.go

```diff
@@ -60,18 +62,60 @@ func (h UploadActionHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request
 	rw.WriteHeader(http.StatusOK)
 }
 
-func (h UploadActionHandler) do(rw http.ResponseWriter, req *http.Request) error {
+func (h ImageHandler) do(rw http.ResponseWriter, req *http.Request) error {
 	vars := mux.Vars(req)
-	action := vars["action"]
-	switch action {
-	case actionUpload:
+	operation := ""
+	if _, found := vars["action"]; found {
+		operation = vars["action"]
+	} else if _, found := vars["link"]; found {
+		operation = vars["link"]
+	}
+	switch operation {
+	case operationUpload:
 		return h.uploadImage(rw, req)
+	case operationDownload:
+		return h.downloadImage(rw, req)
 	default:
 		return apierror.NewAPIError(validation.InvalidAction, "Unsupported action")
 	}
 }
 
-func (h UploadActionHandler) uploadImage(rw http.ResponseWriter, req *http.Request) error {
+func (h ImageHandler) downloadImage(rw http.ResponseWriter, req *http.Request) error {
+	vars := mux.Vars(req)
+	namespace := vars["namespace"]
+	name := vars["name"]
+
+	bkimgName := fmt.Sprintf("%s-%s", namespace, name)
+	downloadUrl := fmt.Sprintf("http://longhorn-backend.longhorn-system:9500/v1/backingimages/%s/download", bkimgName)
+	downloadReq, err := http.NewRequestWithContext(req.Context(), http.MethodGet, downloadUrl, nil)
+	if err != nil {
+		return fmt.Errorf("Failed to create the download request: %w", err)
```

Comment:
suggest to add `bkimgName` into error message for here and following

Reply:
sure, I would improve the error logging.

Reply:
Already change the logging, please check again.